### PR TITLE
WS-125 - Add change nft function to NFTHandler

### PR DESCRIPTION
**Pull request by AgusVelez5**
**Type of MR:**

- [x] Feature
- [ ] Fix
- [ ] HotFix

**Small description:**

- Add change nft function to NFTHandler
- Add NFTData type

**Aditional observations/considerations:**
...

# Checklist:

- [x] My code follows the styleguides of the project (successfully pass linters)
- [x] I tested my code.
- [x] I commented/docummented my code in a way that complex parts are explained.
- [ ] If required, i updated project's README file.
- [x] I added/updated tests that tests my changes in code.
- [x] All previous and new tests pass.
- [ ] I commit any new dependency.
- [ ] I updated UML diagram in confluence if any model was modified in this MR.

### CODE REVIEW

File: packages/nft/src/NFTHandler.ts [hunk 1]
```diff
@@ -117,4 +120,70 @@ export class NFTHandler {
...
+   * @param {string} tokenId - The tokenId of the NFT you want to set.
+   * @param {NFTMetadata} nftMetadata - The metadata you want to set.
+   * @param {string} image - Base64 of the image you want to set.
+   * @returns The transaction object
+   */
+  static async change(
+    nftManager: NFT721Manager,
+    storage: StorageType,
+    nftFormat: NFT_METADATA_FORMAT,
+    nftData: { tokenId: string; image?: string; nftMetadata?: NFTMetadata }
```
Comment (alfarkas) [inline packages/nft/src/NFTHandler.ts:139]: Should the type be an interface?

File: packages/nft/src/NFTHandler.ts [hunk 2]
```diff
@@ -117,4 +120,70 @@ export class NFTHandler {
...
+  ): Promise<interfaces.TransactionResponseExtended | undefined> {
+    const { tokenId, image, nftMetadata } = nftData
+
+    if (!nftManager.isActive) {
+      throw MUST_ACTIVATE
+    }
+
+    if (!nftManager.implements('setTokenURI', ['uint256', 'string'])) {
```
Comment (alfarkas) [inline packages/nft/src/NFTHandler.ts:147]: We should have done this before, but it may be best to make a decorator out of this, by adding [this plugin](https://babeljs.io/docs/en/babel-plugin-proposal-decorators), to improve code reusability and cleanliness. Since is basically always the same. Same goes for isActive.
If possible.

We can do it in a different PR, and update all the places that are doing this at once.

File: packages/nft/src/__tests__/NFTHandler.test.ts [hunk 3]
```diff
@@ -224,3 +225,158 @@ describe('NFTHandler getNFTListByOwner function', () => {
...
+
+    await expect(
+      NFTHandler.change(nftManager, storage, NFT_METADATA_FORMAT.IMAGE, {
+        tokenId,
+        image
+      })
+    ).rejects.toThrow(errors.TRANSACTION_FAILED(fakeError))
+  })
+
+  it('Should return the transaction of the setTokenURI execution (IMAGE)', async () => {
```
Comment (alfarkas) [inline packages/nft/src/__tests__/NFTHandler.test.ts:308]: I think you could combine the following tests by using it.each for the format.

**Review by alfarkas — APPROVED** [review]

File: packages/nft/src/NFTHandler.ts [hunk 4]
```diff
@@ -117,4 +120,68 @@ export class NFTHandler {
...
+   * @param {NFT_METADATA_FORMAT} nftFormat - The format of the NFT metadata.
+   * @param {NFTData} nftData - The data of the NFT you want to set.
+   * @returns The transaction object
+   */
+  static async change(
+    nftManager: NFT721Manager,
+    storage: StorageType,
+    nftFormat: NFT_METADATA_FORMAT,
+    nftData: NFTData
+  ): Promise<interfaces.TransactionResponseExtended | undefined> {
```
Comment (fedeaviles) [inline packages/nft/src/NFTHandler.ts:138]: I think this should be `ExecuteResponse` instead of `TransactionResponseExtended | undefined`. That's what the `execute` returns

File: packages/nft/src/NFTHandler.ts [hunk 5]
```diff
@@ -117,4 +120,68 @@ export class NFTHandler {
...
+        method: 'setTokenURI(uint256,string)',
+        params: [tokenId, tokenUri]
+      })
+
+      return transactionResponse
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      throw TRANSACTION_FAILED(error)
+    }
+  }
```
Comment (fedeaviles) [inline packages/nft/src/NFTHandler.ts:186]: Why don't we just return what the `execute` returns? The `execute` function is already throwing this exception

**Review by alfarkas — APPROVED** [review]

**Review by fedeaviles — APPROVED** [review]
